llvm/test/CodeGen/AMDGPU/llvm.amdgcn.sched.group.barrier.gfx12.ll

```diff
@@ -0,0 +1,333 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc -mtriple=amdgcn -mcpu=gfx1200 -verify-machineinstrs -misched-cluster=0 < %s | FileCheck -check-prefix=GCN %s
```

Comment:
Don't need -verify-machineinstrs

Reply:
This is the 2nd time I see this comment from you. Can you please elaborate why and when something needs it and why and when not? I understand that verifier takes testing time. Anything else?

Reply:
Actually I think this -verify-machineinstrs is needed. 
In llvmbot,  -DLLVM_ENABLE_EXPENSIVE_CHECKS=ON is specified, thus this may seems redundant.
However, in downstream branches, we may not have this cmake option,  -verify-machineinstrs may still help.
Any other reasons?

Reply:
It's brutally expensive. Unless you are adding a test for a case that hits a verifier error, it's probably not worth it. We've traditionally blindly added it to every test. The expensive checks bits will get it eventually if it does fail the verifier

Reply:
> It's brutally expensive. Unless you are adding a test for a case that hits a verifier error, it's probably not worth it. We've traditionally blindly added it to every test. The expensive checks bits will get it eventually if it does fail the verifier

This changes the guidelines. Particularly your own guidelines. I really want to understand where is a borderline between a 'brutally expensive' and normal. I remember the times when every review was down because it was missing this flag. Shall we just omit it always now, because it will eventually fail with expensive checks? I mean, I am not opposed. But there shall be a common policy.

Reply:
As a side note, I really don't like that out target takes so much time to test with debug build comparing to others.

Reply:
The AMDGPU tests in debug are much slower, largely because of excessive use of -verify-machineinstrs. We should be more selective. We never really had a guideline written down, but I think the useful cases are where the patch is fixing a verifier error. All of the intrinsic support cases are not that useful to verify for example